src/client/sandbox/code-instrumentation/location/index.js

```diff
@@ -34,11 +37,17 @@ export default class LocationAccessorsInstrumentation extends SandboxBase {
 
         window[INSTRUCTION.setLocation] = (location, value) => {
             if (isLocation(location)) {
-                location = value;
+                var resourceType = null;
 
-                return location;
-            }
+                if (window.self !== window.top) {
+                    value        = destLocation.resolveUrl(value, window.top.document);
+                    resourceType = stringifyResourceType(true);
```

Comment:
So we need to do the following:
1)rename `stringifyResourceType` to the `getResourceTypeString`
2)move 3 function arguments to the single parameters object